### PR TITLE
SAK-32500 In/New is affected by instructor submission while in "Student View" or in "Enter access View" in the Assignment tool

### PR DESCRIPTION
**Pull request by joaquinmarques**

### CODE REVIEW

**Comment by jonespm** [comment]
I know assignment has a completely local feature for "Student View" and separate templates. When using the role swap this like other tools have a problem because they can't tell if it's a roleswapped (temporary) user or someone with another role.

The thing I really don't like about this fix is it *looks* like you'd have the same issue for any role that isn't instructor that swaps to a student role. (For a simple example maintain->access in a project site but other schools may make custom roles)

I feel like there should be some API call that would return if a users effective role matches their role in the site. In portal (SiteHandler-includeTabs) it looks like it does this check inline but this generic code seems useful and might be safer even in this case to do that similar check.

**Comment by joaquinmarques** [comment]
In order to know if the user is instructor I am checking if the user has the "asn.revise" or "asn.new" permission, so custom roles are independent. Is it correct?

**Comment by steveswinsburg** [comment]
Fyi GradebookNG  has a business service method to check if a user is role
swapped.  This could go to kernel.

sent from my mobile device

**Comment by jonespm** [comment]
That sounds great Steve, thanks for the tip! I'll work on moving that into kernel and having portal also use it. I'd rather this patch also use this method as it should simplify the code here.

File: assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 1]
```diff
@@ -385,6 +385,10 @@
 	/** ******************** student's view assignment submission ****************************** */
 	/** the assignment object been viewing * */
 	private static final String VIEW_SUBMISSION_ASSIGNMENT_REFERENCE = "Assignment.view_submission_assignment_reference";
+	
+	/** the assignment object been viewing by instructor * */
+	/** SAK-32500 **/
```
Comment (mpellicer) [inline assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: Please remove Jira reference

File: assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 2]
```diff
@@ -5541,6 +5545,33 @@ public void doView_submission(RunData data)
 		}
 
 		String submitterId = params.get("submitterId");
+		
+		// SAK-32500 START
```
Comment (mpellicer) [inline assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: Please remove Jira reference

File: assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 3]
```diff
@@ -5541,6 +5545,33 @@ public void doView_submission(RunData data)
...
+			submitterIdInstructor = "instructor";
+		}		
+		// From enter as student link chef_assignments_list_assignments.vm END
+				
+		if ("instructor".equals(submitterIdInstructor)) 
+		{
+			state.setAttribute(VIEW_SUBMISSION_ASSIGNMENT_INSTRUCTOR, submitterIdInstructor);
+		}
+		
+		// SAK-32500 END
```
Comment (mpellicer) [inline assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: Please remove Jira reference

File: assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 4]
```diff
@@ -6390,6 +6421,7 @@ private void post_save_submission(RunData data, boolean post)
 		String contextString = (String) state.getAttribute(STATE_CONTEXT_STRING);
 		String aReference = (String) state.getAttribute(VIEW_SUBMISSION_ASSIGNMENT_REFERENCE);
 		Assignment a  = getAssignment(aReference, "post_save_submission", state);
+		String assignmentId = ""; // SAK-32500
```
Comment (mpellicer) [inline assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: Please remove Jira reference

File: assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 5]
```diff
@@ -6433,6 +6465,13 @@ private void post_save_submission(RunData data, boolean post)
 				}
 			}
 
+			// SAK-32500 START
```
Comment (mpellicer) [inline assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: Please remove Jira reference

File: assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 6]
```diff
@@ -6433,6 +6465,13 @@ private void post_save_submission(RunData data, boolean post)
...
+			String instructor = null;
+			instructor = (String) state.getAttribute(VIEW_SUBMISSION_ASSIGNMENT_INSTRUCTOR);			
+			if(instructor == null)
+			{
+			// SAK-32500 END
```
Comment (mpellicer) [inline assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: Please remove Jira reference

File: assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 7]
```diff
@@ -6847,14 +6886,16 @@ private void post_save_submission(RunData data, boolean post)
 				} // if-else
 	
 			} // if
+			
+			} // if - SAK-32500
```
Comment (mpellicer) [inline assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: Please remove Jira reference

File: assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 8]
```diff
@@ -6847,14 +6886,16 @@ private void post_save_submission(RunData data, boolean post)
...
 	
 			if (state.getAttribute(STATE_MESSAGE) == null)
 			{
 				state.setAttribute(STATE_MODE, MODE_STUDENT_VIEW_SUBMISSION_CONFIRMATION);
 			}
             LearningResourceStoreService lrss = (LearningResourceStoreService) ComponentManager
                     .get("org.sakaiproject.event.api.LearningResourceStoreService");
-            if (null != lrss) {
+            if (null != lrss && !"".equals(assignmentId)) { // SAK-32500
```
Comment (mpellicer) [inline assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: Use StringUtils.isEmpty instead.

File: assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 9]
```diff
@@ -12746,6 +12787,7 @@ private void resetViewSubmission(SessionState state)
 		state.removeAttribute(VIEW_SUBMISSION_TEXT);
 		state.setAttribute(VIEW_SUBMISSION_HONOR_PLEDGE_YES, "false");
 		state.removeAttribute(GRADE_GREATER_THAN_MAX_ALERT);
+		state.removeAttribute(VIEW_SUBMISSION_ASSIGNMENT_INSTRUCTOR); // SAK-32500
```
Comment (mpellicer) [inline assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: Please remove Jira reference

**Comment by joaquinmarques** [comment]
Removed Jira references

**Comment by jonespm** [comment]
The method from KNL-1529 b1aa9d3590f41b4e701fd93aaafd0fb46130587e securityService.isUserRoleSwapped() is added now so it should eliminate most of the custom code here in AssignmentAction.java.

**Pull request closed without merging**